lib/plugins/inspec-habitat/lib/inspec-habitat/profile.rb

```diff
@@ -82,179 +99,146 @@ def upload
         )
       end
 
-      def delete_work_dir
-        @log.debug("Deleting work directory #{work_dir}")
-        FileUtils.rm_rf(work_dir) if Dir.exist?(work_dir)
+      private
+
+      def create_working_dir
+        working_dir = Dir.mktmpdir
+        @log.debug("Generated working directory #{working_dir}")
+        working_dir
       end
 
-      def setup
-        @log.info("Setting up profile at #{path} for Habitat...")
-        create_profile_object
-        verify_profile
-        vendor_profile_dependencies
-        create_habitat_directories(path)
-        create_plan(path)
-        create_run_hook(path)
-        create_default_config(path)
+      def duplicate_profile(path, working_dir)
+        profile = profile_from_path(path)
+        copy_profile_to_working_dir(profile, working_dir)
+        profile_from_path(working_dir)
       end
 
-      private
+      def prepare_profile!(profile)
+        vendored_profile = vendor_profile_dependencies!(profile)
+        verify_profile(vendored_profile)
+        setup(vendored_profile)
+      end
 
-      def create_profile_object
-        @profile = Inspec::Profile.for_target(
+      def profile_from_path(path)
+        Inspec::Profile.for_target(
           path,
           backend: Inspec::Backend.create(Inspec::Config.mock),
         )
       end
 
-      def verify_profile
+      def copy_profile_to_working_dir(profile, working_dir)
+        @log.info('Copying profile contents to the working directory...')
+        profile.files.each do |profile_file|
+          next if File.extname(profile_file) == '.hart'
+
+          src = File.join(profile.root_path, profile_file)
+          dst = File.join(working_dir, profile_file)
+          if File.directory?(profile_file)
+            @log.debug("Creating directory #{dst}")
+            FileUtils.mkdir_p(dst)
+          else
+            @log.debug("Copying file #{src} to #{dst}")
+            FileUtils.cp_r(src, dst)
+          end
+        end
+      end
+
+      def verify_profile(profile)
         @log.info('Checking to see if the profile is valid...')
```

Comment:
Personal style preference here. I avoid putting positive confirmation messages at `info` level and instead prefer them at `debug` level.

IMO, `info` level logs should include data like metrics - its a level that could be parsed by tooling to interpret the health of the tool. `warn` is for alerting monitors that a service is going to fail if someone does not intervene soon and `debug` (or `trace`) is for trying to trace through a program. So I would think this is a `debug` level message - it is communicating that the tool is functioning as intended, and if something goes wrong the `exit_with_error` will already communicate the issue to the user.

Reply:
Nope, seems valid to me. I'll make that change.

Reply:
I agree, should be fixed now.